src/enum_set.h

```diff
@@ -0,0 +1,46 @@
+#pragma once
+
+#include <bitset>
+
+template<typename T>
+class EnumSet {
```

Comment:
Ok I like this idea! I think we may even want to add bitwise operator overloads later.